langchain/src/embeddings/ollama.ts

```diff
@@ -95,7 +95,17 @@ export class OllamaEmbeddings extends Embeddings {
   async _request(prompt: string): Promise<number[]> {
     const { model, baseUrl, requestOptions } = this;
 
-    const response = await fetch(`${baseUrl}/api/embeddings`, {
+    let formattedBaseUrl = baseUrl;
+    if (formattedBaseUrl.startsWith("http://localhost:")) {
+      // Node 18 has issues with resolving "localhost"
+      // See https://github.com/node-fetch/node-fetch/issues/1624
+      formattedBaseUrl = formattedBaseUrl.replace(
```

Comment:
Just to reduce the amount of magic, can we do something like this and change the default in the constructor instead?

https://github.com/langchain-ai/langchainjs/blob/main/langchain/src/chat_models/ollama.ts#L35

Reply:
I was about to change the baseUrl to 127.0.0.1 but after looking through the rest of the Ollama code, I realized that all the other Ollama functions like `ChatOllama` and `Ollama` class call on this `createOllamaStream` which has this reformatting logic

```ts
  let formattedBaseUrl = baseUrl;
  if (formattedBaseUrl.startsWith("http://localhost:")) {
    // Node 18 has issues with resolving "localhost"
    // See https://github.com/node-fetch/node-fetch/issues/1624
    formattedBaseUrl = formattedBaseUrl.replace(
      "http://localhost:",
      "http://127.0.0.1:"
    );
  }
```
[Source](https://github.com/langchain-ai/langchainjs/blob/d048313543d8cff5acc91bcb5ee1c6defa830cf5/langchain/src/util/ollama.ts#L96)  

I thought this was more appropriate 
- So the default baseUrl across Ollama classes can be consitently `localhost` - so the docs are consistent too in terms of showing the default baseUrl
- The end user won't mess it up if they override the default baseUrl with `localhost` either

Of course, I can just update the baseUrl in the `OllamaEmbeddings` to be 127.0.0.1 as first discussed if you desire. The docs will just be updated as well to 127.0.0.1 as default and will be inconsistent with the rest of the Ollama classes but once you confirm either approach, I'll make the PR.